FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,40 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case let .success((data, response)):
+				guard response.statusCode == 200, let items = try? JSONDecoder().decode(ImageMapper.ImageItems.self, from: data) else {
+					completion(.failure(Error.invalidData))
+					return
+				}
+				completion(.success(items.images))
```

Comment:
Since there is the `ImageMapper` helper for performing mappings you can create a function and move all the mapping logic to it as well:

```swift
private struct ImageMapper {

	...
	
	static func map(_ data: Data, response: HTTPURLResponse) -> FeedLoader.Result {
		guard response.statusCode == 200,
		      let items = try? JSONDecoder().decode(ImageItems.self, from: data) else {
			return .failure(RemoteFeedLoader.Error.invalidData)
		}

		return .success(items.images)
	}
}
```

Then the calling site would be:

```suggestion
				completion(ImageMapper.map(data, response: response))
```